### PR TITLE
WIP: Add initial tabs research

### PR DESCRIPTION
**Pull request by bkardell**
Adding initial tabs research modeled in part after slider research model/link up and based on shared work we did during gathering in a gdoc.

Note that this uses images currently loaded from https://lh4.googleusercontent.com/ and I'm sure we'll have to fix that but given the troubles I've had I thought that having a WIP: PR where perhaps people could give more specific help/advice on sorting that out would be more efficient.

### CODE REVIEW

**Review by davatron5000 — COMMENTED** [review]
Great writeup. I have a few nitpicks/formatting things here. 

Related, but not related to this, I'm wondering about the readability of these docs. It's a beefy research document, so it is what it is, but I'm just mulling around general ideas to eliminate the wall of text vibe (for future research going forward as well)

- I like your illustration idea you noted
- We could list all the "features" and "sub-features" and do a compatability matrix style table for each design system or implementation. It could be really generic "full", "partial", or "no" support for each future. 
- Hide screenshot tables in a details/summary?

Honestly tho, the thing I maybe want most is some `<feature>` styling:

```css
div > figure {
  font-style: italic;
  text-align: center;
  max-width: 80%;
  margin: 1.25rem auto;
}

figure img {
  margin: auto
}
```

Maybe I can pull the Gatsby site down and make a PR separate for those things. 🤷

File: research/src/pages/tabs.research.parts.mdx [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+Atlaskit |Not explicitly supported, the API for providing a label is a string. 
+Ant Design | ![Ant design tabs with an icon and text]()
+Carbon Design | Not explicitly supported
+Fast | Not explicitly supported
+Lightning Components | ![something](https://lh3.googleusercontent.com/qiG1YSwFv7J3IQSZkjux5K1J7WnDL8aGdUHD1BO6uHgUCGMRQG6E8ngN30d6vHDyZaMUaniJR-JiEXQJag9bwB-4OFLGELWuHKx3Uz567UaNz6eY4OOrLKFk55KxS-n7PfzpBcvG)
+Material | ![Material design with icon and text in horizontal top position](https://lh5.googleusercontent.com/Yof7dEFxA068PtyzK7OpXhjfjOZA9PHezerZUjMNA-EsSTNY2Q2AZMNQLk9Oo_LBlnZ5BrQAVNeZR1j1BaLc8noIniKNmxfftrFHFgDuXBVZzIJGkBa_UuoFuGa2jugWwwgTlM0u)
+Oragami | Not specifically supported, but markup based and seems to support embedding custom content
+Semantic UI | Not explicitly supported ('menus' separate from the component they call 'tabs' which is more of a set of mutually exclusive selectable sections, but looks like these menus can contain embedded custom content)
+Shoelace | (not specifically supported, but seems to support embedding custom content)
+Vaadin | (not specifically supported, but explicitly supports embedding custom content)
```
[the remark this answers is not in the document]
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx:81]: Happy to do what you like, I thought we'd sorted all of this in the gdoc.  My only concern with  removing the notes is that the subtleties actually do matter - we're often not comparing apples to apples here.  But... suggest something?

File: research/src/pages/tabs.research.parts.mdx [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+    </figcaption>
+</figure>
+
+
+Most design systems recommend homogeneity in the same Tab Bar (that is, against mixing modes of combination of text/icon). Some actively enforce this, via API, some don't.
+
+
+#### Directionality/Mirroring
+There are some well established expectations with regard to internationalization for certain parts of Tab Sets.  In right-to-left languages, for example, generally speaking some kind of RTL mirroring applies - however, it will work work differently depending on a number of other factors.  Some of these are very well established, others less so.  For example, when tabs are grouped in the in the horizontal top position, in RTL the order of the tabs is reversed, and the icon/text flip too.  Keyboard controls, however, continue to maintain spatial order (that is, pressing 'left arrow' will move selection to the tab displayed to the left of the currently selected and so on. First and last shortcuts will be impacted as per the directionality.  As systems add additional elements or orientations this becomes slightly less clear.
```
Comment (gregwhitworth) [inline research/src/pages/tabs.research.parts.mdx:97]: `work work differently` -> `work differently`

File: research/src/pages/tabs.research.parts.mdx [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+<figure>
+  <img src="https://lh4.googleusercontent.com/SZMUPImNdUKl-TL7z_RBsNIqj2tcofFM2IjGV2qa4025hsAPylWVeAXC04SmeOc2JwIT-6kimj_VObEhJ4YSxSA3OjxYZkHpaTsFP97izDxwW_Y0pwfC_EAo9CBSvXj2QbZQ4yhS" />
+  <figcaption>Material design with icon and text in horizontal top position in LTR, and the same in RTL</figcaption>
+
+</figure>
+
+Mirroring should work similarly in horizontal bottom positioning.  However, if left and right positioned tabs are supported, they will reverse the order of their alignment and their icon/text relationship, but not the order of the tabs themselves.
+
+
+##### Icon Alignment
```
Comment (gregwhitworth) [inline research/src/pages/tabs.research.parts.mdx:111]: Should this be a `####` heading? It seems small compared to Directionality/Mirroring heading and it is hard to tell it's a subheading. If this is a subsection of Directionality/Mirroring then this may be another example of @davatron5000 recommending some styling adjustments to aid in these larger research documents.

File: research/src/pages/tabs.research.parts.mdx [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+
+
+
+ Note: Some can switch any (automatically or via attribute) to ungrouped/inline.
+
+
+### Additional/new types
+Just as over time we have expanded many common definitions to include labels grouped on any of four sides, it is reasonable to expect that new ideas will continue to explore new visualizations of which require similar qualities in terms of necessarily semantic parts, required interactions, events, etc. 
+
+What follows are some interesting examples which can be informative.  Some of these have been developed in various systems specifically "as" a flexible Tab Set interface, while others things break down into parts leaving open the possibility that they could be.  Implementation details aside, these components, which fit the description above are developing and not uncommon entirely uncommon thanks to pressures from things like web interfaces providing for IoT, video games and even the needs of simple responsive design.
```
Comment (gregwhitworth) [inline research/src/pages/tabs.research.parts.mdx:221]: `uncommon entirely uncommon` -> `not uncommon`

File: research/src/pages/tabs.research.parts.mdx [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+
+What follows are some interesting examples which can be informative.  Some of these have been developed in various systems specifically "as" a flexible Tab Set interface, while others things break down into parts leaving open the possibility that they could be.  Implementation details aside, these components, which fit the description above are developing and not uncommon entirely uncommon thanks to pressures from things like web interfaces providing for IoT, video games and even the needs of simple responsive design.  
+
+Given that we are looking at components in systems with diverse names and qualities it is difficult to discuss how wide spread this really is.  The support and quality varies from not supporting a component like this at all, to providing a component that looks like this but has different qualities, to providing a different component that looks like this and has the same qualities, but cannot be visualized as grouped tabs easily, to responsive components that respond and switch their types fluidly.  As such, we have simply provided examples and explanations of these types.
```
Comment (gregwhitworth) [inline research/src/pages/tabs.research.parts.mdx:223]: Nit: I think this paragraph could be removed or reduced to a single sentence, it's kind of hard to follow in general.

File: research/src/pages/tabs.research.parts.mdx [hunk 6]
```diff
@@ -0,0 +1,439 @@
...
+
+#### Ungrouped Vertical (block sections)
```
Comment (gregwhitworth) [inline research/src/pages/tabs.research.parts.mdx:225]: this is a beautiful example of investigating how RWD may/may not impact anatomy/states/behaviors - great work

File: research/src/pages/tabs.research.parts.mdx [hunk 7]
```diff
@@ -0,0 +1,439 @@
...
+<figure>
+  <img src="https://lh3.googleusercontent.com/z_QukHGXEFyf6yc0N9wl9SjAkida7dgmp1cnw5WlcgmTBDYlSRsqEErlZeug_IoLBMHAC2gXZkqBj9vQQ5vP7muj0SdJlhrVrt79o3yO1rNQnBHQN-K4rHlZQ1qwRzPGWxcb3Ln3" />
+  <img src="https://lh6.googleusercontent.com/t0jeYGxYYMSVVgOCrD4OZbUt6OyvmdLy88isZcifl8XKljv9nzTDD01GlpS9pRJKbDd6VIpgGm4UiAsKbRG6_F2WDmnIa2fLY_P64ey1Z7doy0RC4U_Ptfiy6QCt_BpLfOL5CPeF" />
+  <figcaption>Vaadin overflows the TabList with a vertical scroll bar when grouped vertically, and with interactive next/previous buttons which advance the clipped scroll area by one page when grouped vertically.</figcaption>
+</figure>
+
+
+
+## Visual transitions
+A rarely occurring enhancement, but not entirely uncommon and seemingly growing in popularity is providing a visual transition effect between the finite states of tabs.  It it worth consider a feature on its own in the sense that there are several implementations which are built in such a way that this is not possible to achieve.
```
Comment (gregwhitworth) [inline research/src/pages/tabs.research.parts.mdx:338]: `It it worth consider` -> `It is worth considering...`

File: research/src/pages/tabs.research.parts.mdx [hunk 8]
```diff
@@ -0,0 +1,439 @@
...
+
+Note that if tabsets are 'dismissable' they can also be 'entirely dismissable' (the entire control disappears) or they have to deal with what happens when only one tab remains.  It seems the choice to support this, not support it, or how to deal with whether they are individually, as a whole, or only as long as more than one remains varies quite a bit, though realistically without 2 (that are not disabled) it gets more difficult to really call them "tabs", thus many design systems at least recommend, if not require  minimum of two.  In many windowing toolkits, "dismissable" cases are dealt with extensions to a common interface and not called 'tabs', or 'tabs' has a narrower definition in one direction or the other.
+ 
+ This might be thought of to go hand in hand with the ability to add a tab, and while not uncommon in programmatic APIs, it is considerably more rare from a user interface point of view and there aren't plentiful ideas in practice from libraries to draw from.
```
Comment (gregwhitworth) [inline research/src/pages/tabs.research.parts.mdx:381]: Just an FYI, we have this all throughout Salesforce UI.

**Review by gregwhitworth — APPROVED** [review]
I love this and as we've discussed in the past - the research section is a scratch pad and should always be evolving as long as it has some coherence in the direction which this certainly does. @davatron5000 has some solid feedback on how to break up the wall of text and some styling but I'd recommend them in a seperate PR and enable the group to start discussing concrete issues. There are many while I was reading this I wanted to provide thoughts on but this isn't the place for that :)

Great work by everyone that had a hand in this research, I know it's not easy and takes a LOT OF time; it is greatly appreciated.

File: research/src/pages/tabs.research.parts.mdx (hunk 2 above)
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx]: corrected

File: research/src/pages/tabs.research.parts.mdx (hunk 3 above)
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx:111]: Yes it is a subheading/subsection - I agree with @davatron5000 that this could use style help

File: research/src/pages/tabs.research.parts.mdx (hunk 4 above)
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx]: updated

File: research/src/pages/tabs.research.parts.mdx (hunk 5 above)
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx]: Tried again, please see if this reads better?

File: research/src/pages/tabs.research.parts.mdx (hunk 6 above)
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx:223]: ty, I hope I didn't screw it up with my rephrasing for the previous point

File: research/src/pages/tabs.research.parts.mdx (hunk 7 above)
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx]: updated

File: research/src/pages/tabs.research.parts.mdx (hunk 8 above)
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx:379]: sorry it's hard to tell what the 'this' is here - can you be more specific?

File: research/src/pages/tabs.research.parts.mdx [hunk 9]
```diff
@@ -0,0 +1,437 @@
...
+Tab Labels identify the contents of a Tab Panel, creating a conceptual Tab Section.  They are semantically, and interactively linked.
+
+The most fundamental aspect of a Tab Label is text, though some systems support additional content inside the label (and some actively prevent it via API design).  The most common variant includes an icon, though support is very mixed on how this is supported (or prevented).  There are also implications of these choices on things like internationalization and accessibility which are covered later.
+
+#### Tab Labels with Icons support
+
+System | Screenshot/Notes  
+---|---|
+Atlaskit |Not explicitly supported, the API for providing a label is a string. 
+Ant Design | ![Ant design tabs with an icon and text]()
```
[the remark this answers is not in the document]
Comment (bkardell) [reply research/src/pages/tabs.research.parts.mdx]: added, thanks

**Review by una — APPROVED** [review]
